pennylane_lightning/core/src/simulators/lightning_tensor/cutn/CutnBase.hpp

```diff
@@ -0,0 +1,227 @@
+// Copyright 2024 Xanadu Quantum Technologies Inc.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+
+//     http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @file CutnBase.hpp
+ * Base class for cutensornet backend.
+ */
+
+#pragma once
+
+#include <complex>
+#include <type_traits>
+#include <vector>
+
+#include <cuda.h>
+#include <cutensornet.h>
+
+#include "CudaTensor.hpp"
+#include "TensorBase.hpp"
+#include "TensornetBase.hpp"
+#include "cuda_helpers.hpp"
+#include "cutnError.hpp"
+#include "cutn_helpers.hpp"
+
+/// @cond DEV
+namespace {
+namespace cuUtil = Pennylane::LightningGPU::Util;
+using namespace Pennylane::LightningGPU;
+using namespace Pennylane::LightningTensor::Cutn;
+using namespace Pennylane::LightningTensor::Cutn::Util;
+} // namespace
+///@endcond
+
+namespace Pennylane::LightningTensor::Cutn {
+
+template <class Precision, class Derived>
+class CutnBase : public TensornetBase<Precision, Derived> {
+  protected:
+    bool MPSInitialized_ = false;
+    bool MPSFinalized_ = false;
```

Comment:
Does it make sense to define an enum class to specify the status of the MPS instance at any time? It will remove these booleans and create some relations between different stages.